src/Transports/MassTransit.KafkaIntegration/KafkaIntegration/KafkaMessageConsumer.cs

```diff
@@ -30,14 +31,18 @@ public KafkaMessageConsumer(ReceiveSettings receiveSettings, KafkaReceiveEndpoin
         {
             _receiveSettings = receiveSettings;
             _context = context;
+            _limit = new SemaphoreSlim(receiveSettings.ConcurrentMessageLimit);
 
             _cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(Stopping);
             _checkpointTokenSource = CancellationTokenSource.CreateLinkedTokenSource(Stopped);
             _lockContext = new ConsumerLockContext(consumerContext, receiveSettings, _checkpointTokenSource.Token);
 
             _consumer = consumerContext.CreateConsumer(this, HandleKafkaError);
 
-            _executorPool = new CombinedChannelExecutorPool(_lockContext, receiveSettings);
+            IHashGenerator hashGenerator = new Murmur3UnsafeHashGenerator();
+            _executorPool = new PartitionChannelExecutorPool<ConsumeResult<byte[], byte[]>>(x => x.Message.Key, hashGenerator,
```

Comment:
does x.Message.Key properly handle null??

Reply:
we do not have nulls, it is always empty array